File: ui/src/Pages/AdminPanel/Category/Category.js
import React from 'react'
import { Datagrid, TextField, List } from 'react-admin'

const Category = () => {
  return (
    <List>
      <Datagrid>
        <TextField disabled source='id' />
        <TextField source='name' />
        <TextField source='code' />
        <TextField source='description' />
      </Datagrid>
    </List>
  )
}

export default Category

File: backend/UrbanAisle/src/main/java/com/urbanaisle/store/entities/Category.java
package com.urbanaisle.store.entities;

import java.util.List;
import java.util.UUID;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name= "categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category {
	
	@Id
	@GeneratedValue
	private UUID id;
	
	@Column(nullable = false)
	private String name;
	
	@Column(nullable = false)
	private String code;
	
	@Column(nullable = false)
	private String description;
	
	@OneToMany(mappedBy = "category", cascade = CascadeType.ALL)
	private List<CategoryType> categoryTypes;
	
}
